src/rospkg/rospack.py

```diff
@@ -204,40 +211,53 @@ def get_path(self, name):
         else:
             return self._location_cache[name]
 
-    def _load_manifest(self, name):
+    def _load_manifest(self, name, typeof_depend=0):
         """
         :raises: :exc:`ResourceNotFound`
         """
-        retval = self._manifests[name] = parse_manifest_file(self.get_path(name), self._manifest_name, rospack=self)
+        retval = self._manifests[name] = parse_manifest_file(self.get_path(name), self._manifest_name, rospack=self, typeof_depend=typeof_depend)
         return retval
 
-    def get_depends(self, name, implicit=True):
+    def get_depends(self, name, implicit=True, typeof_depend=0):
         """
         Get dependencies of a resource.  If implicit is ``True``, this
         includes implicit (recursive) dependencies.
 
         :param name: resource name, ``str``
         :param implicit: include implicit (recursive) dependencies, ``bool``
-
+        :param typeof_depend:
+                0: all kinds of depend.
+                1: only runtime_depend, such as run_depend in package.xml format 1 and 2.
+                2: only build_depend.
+        :type typeof_depend: int
         :returns: list of names of dependencies, ``[str]``
         :raises: :exc:`InvalidManifest` If resource or any of its
           dependencies have an invalid manifest.
         """
+        # When typeof_depend flag is different from the one previously used,
+        # flag, re-compute the dependency.
+        if not self._typeof_dependency == typeof_depend:
+            self._typeof_dependency = typeof_depend
+            # Resetting cache. This may not be necessary when the duplicate invocation
+            # issue is addressed https://github.com/ros-infrastructure/rospkg/pull/129#issuecomment-364572565
```

Comment:
Can you please elaborate on this. How would the fix from 129 address this? Assuming 129 gets merged how do you expect this patch to change?